src/NHibernate/Logging.log4net.cs

```diff
@@ -12,16 +13,48 @@ namespace NHibernate
 	public class Log4NetLoggerFactory: ILoggerFactory
 #pragma warning restore 618
 	{
-		private static readonly System.Type LogManagerType = System.Type.GetType("log4net.LogManager, log4net");
+		internal static readonly Assembly Log4NetAssembly;
+		private static readonly System.Type LogManagerType;
 		private static readonly Func<Assembly, string, object> GetLoggerByNameDelegate;
 		private static readonly Func<System.Type, object> GetLoggerByTypeDelegate;
 
 		static Log4NetLoggerFactory()
 		{
+			LogManagerType = GetLogManagerType();
+			if (LogManagerType == null)
+				return;
+
+			Log4NetAssembly = LogManagerType.Assembly;
 			GetLoggerByNameDelegate = GetGetLoggerByNameMethodCall();
 			GetLoggerByTypeDelegate = GetGetLoggerMethodCall<System.Type>();
 		}
 
+		public Log4NetLoggerFactory()
+		{
+			if (LogManagerType == null)
+				throw new TypeLoadException("Cannot find log4net.LogManager type");
+		}
+
+		// Code adapted from ReflectHelper.TypeFromAssembly, which cannot be called directly due
+		// to depending on the logger.
+		private static System.Type GetLogManagerType()
+		{
+			var typeName = new AssemblyQualifiedTypeName("log4net.LogManager", "log4net");
+			// Try to get the type from an already loaded assembly
+			var type = System.Type.GetType(typeName.ToString());
```

Comment:
Together with the previous suggestion
```suggestion
			var type = System.Type.GetType(typeName);
```